src/datasource.ts

```diff
@@ -385,6 +392,149 @@ export default class CogniteDatasource extends DataSourceApi<
   fetchSingleAsset = (id: IdEither) => {
     return fetchSingleAsset(id, this.connector);
   };
+  async getRelationshipsDropdowns(
+    refId: string,
+    selector
+  ): Promise<RelationshipsSelectableValue[]> {
+    const settings = {
+      method: HttpMethod.POST,
+      data: {
+        filter: {},
+        limit: 1000,
+      },
+    };
+    try {
+      const response = await this.connector.fetchItems({
+        ...settings,
+        path: `/${selector.type}/list`,
+      });
+      return response.map(({ name, ...rest }) => ({
+        value: rest[selector.keyPropName],
+        label: name,
+      }));
+    } catch (error) {
+      handleError(error, refId);
+      return [];
+    }
+  }
+
+  fetchRelationshipsTargets = async (
+    targets: CogniteQuery[],
+    [min, max]
+  ): Promise<MutableDataFrame[]> => {
+    return Promise.all(
+      targets.map(async (target) => {
+        const { tab, refId, relationsShipsQuery } = target;
+        if (tab === Tab.Relationships) {
+          const { labels, dataSetIds, isActiveAtTime } = relationsShipsQuery;
+
+          const timeFrame = isActiveAtTime && { activeAtTime: { max, min } };
+          const realtionshipsList = await fetchRelationships(
+            {
+              ...filterLabels(labels),
+              ...filterdataSetIds(dataSetIds),
+              ...timeFrame,
+            },
+            this.connector
+          );
+          return !isEmpty(realtionshipsList)
+            ? this.createRelationshipsNode(realtionshipsList, refId)
+            : [];
+        }
+        return [];
+      })
+    ).then((res) => res[0]);
+  };
+
+  createRelationshipsNode = (realtionshipsList, refId): Promise<MutableDataFrame[]> => {
```

Comment:
this function is too large for this file, move it to its own file `cdf/relationships.ts`